### PR TITLE
feat(rlt): migrate OpenPI to JAX-aligned PyTorch Pi0.5

### PR DESCRIPTION
**Pull request by TihayaKousaka**
### Description

This PR migrates RLT OpenPI usage from the old official OpenPI PyTorch path to RLinf's vendored `openpi_pytorch` Pi0.5 implementation, which is aligned with the JAX OpenPI reference.

Main changes:
- Use `openpi_pytorch` for RLT Stage1 SFT and Stage2 feature-model loading.
- Preserve the existing RLT Stage1 data behavior by keeping the legacy OpenPI dataloader path for `openpi_pytorch + use_rlt`.
- Support loading Stage1 FSDP checkpoints directly from the `actor` checkpoint directory so Stage2 can consume the full wrapper checkpoint, including `rlt_module.*`.
- Add RLT support to the `openpi_pytorch` SFT/eval wrappers for Stage1 loss computation and Stage2 feature extraction.
- Update RLT documentation to clarify the `openpi_pytorch` migration, checkpoint path, and Stage1/Stage2 config requirements.

### Motivation and Context

The previous RLT path used the old official OpenPI PyTorch implementation, whose training behavior is weaker than the JAX reference. RLinf now has a vendored `openpi_pytorch` implementation aligned with the JAX OpenPI reference, so RLT should use this path as well.

In RLT experiments, the new path shows faster SFT convergence. The RL token loss also converges faster and reaches a lower value than before, which improves the Stage1 feature model quality. Preliminary Stage2 results are better with the new Stage1 checkpoint.

This migration is intended to be transparent to the RLT algorithm itself: the SFT/model implementation changes, while the RLT Stage2 algorithm and observation/action semantics remain unchanged.

### How has this been tested?

- Verified the RLT Stage1/Stage2 migration path on ManiSkill.
- Verified that Stage1 SFT loss converges faster with the new `openpi_pytorch` path.
- Verified that RL token loss converges faster and reaches a lower final value.
- Verified that Stage2 can consume the new Stage1 checkpoint and shows improved performance.
image
The **green** line is the result of **openpi_rlinf**

image
The **orange** line is the result of **openpi_rlinf**

### Additional information (optional, e.g., figures and logs):

The new Stage1 checkpoint should be passed to Stage2 as the FSDP `actor` directory, for example:

```yaml
rollout:
  rlt_feature_model:
    model_type: "openpi_pytorch"
    model_path: /path/to/stage1/checkpoints/global_step_<step>/actor
```

Do not convert the RLT Stage1 checkpoint to a bare `model.safetensors` for Stage2, because the RLT token module is stored in the full wrapper checkpoint.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: rlinf/models/embodiment/openpi_rlinf/__init__.py [hunk 1]
```diff
@@ -21,18 +21,160 @@
```
Comment (qurakchin) [inline rlinf/models/embodiment/openpi_rlinf/__init__.py:28]: please add a utils/rlt_utils.py

Comment (TihayaKousaka) [reply rlinf/models/embodiment/openpi_rlinf/__init__.py:28]: I have updated it and extracted rlt from this file.

**Review by andylin-hao — APPROVED** [review]

**Review by guozhen1997 — APPROVED** [review]

File: examples/embodiment/config/maniskill_rlt_stage2_ac_mlp.yaml [hunk 2]
```diff
@@ -243,24 +243,31 @@ rollout:
     lora_rank: 32
 
   rlt_feature_model:
-    model_type: "openpi"
+    model_type: "openpi_rlinf"
     model_path: "/path/to/maniskill_rlt_stage1_sft_openpi_pi05/checkpoints/global_step_<step>/actor"
-    precision: null
+    precision: bf16
```
Comment (qurakchin) [inline examples/embodiment/config/maniskill_rlt_stage2_ac_mlp.yaml:248]: The actor.model.precision is fp32, it means this case uses mixed precision training. Have you tested that using bf16 here may affect the reward?

File: rlinf/models/embodiment/openpi_rlinf/transforms_pipeline.py [hunk 3]
```diff
@@ -60,25 +60,34 @@ def build_openpi_transforms(
...
-            norm_stats_path.parent if norm_stats_path.is_file() else norm_stats_path
-        )
-        # ``load_norm_stats`` expects the parent of ``asset_id``; the config
-        # override above has already set ``asset_id`` to this directory name.
-        stats_dir = str(norm_stats_dir_path.parent)
-    else:
-        stats_dir = download.maybe_download(str(model_path))
-    norm_stats = _checkpoints.load_norm_stats(stats_dir, asset_id)
+    norm_stats = data_config.norm_stats
+    if norm_stats is None:
```
Comment (qurakchin) [inline rlinf/models/embodiment/openpi_rlinf/transforms_pipeline.py:74]: add a comment to the function to describe the new priority:
data_config.norm_stats > norm_stats_dir > explicit_norm_stats_path > download

File: examples/embodiment/config/maniskill_rlt_stage2_ac_mlp.yaml [hunk 4]
```diff
@@ -243,24 +243,31 @@ rollout:
...
     num_action_chunks: 10
     action_dim: 8
```
Comment (qurakchin) [inline examples/embodiment/config/maniskill_rlt_stage2_ac_mlp.yaml]: it's better to use ${actor.model.xxx}.

**Review by qurakchin — APPROVED** [review]

**Comment by qurakchin** [comment]
add a comment to describe which color is the openpi pytorch and which one is the openpi rlinf

File: examples/embodiment/config/maniskill_rlt_stage2_ac_mlp.yaml (hunk 2 above)
Comment (TihayaKousaka) [reply examples/embodiment/config/maniskill_rlt_stage2_ac_mlp.yaml:248]: rlt_feature_model is a separate frozen Stage1 feature extractor, not the fp32 Stage2 actor being trained. The previous official OpenPI loader also explicitly converted selected model parameters to bf16 even when precision was null, so null did not mean fp32 inference.  
The vendored openpi_rlinf implementation uses bf16 explicitly, consistent with its Pi0.5 configuration. Its outputs (z_rl, proprio, and ref_chunk) are cast back to fp32 before being consumed by the fp32 Stage2 actor. Therefore, actor optimization remains fp32. We have also run the Stage2 experiment with this setting successfully.

File: rlinf/models/embodiment/openpi_rlinf/transforms_pipeline.py (hunk 3 above)
Comment (TihayaKousaka) [reply rlinf/models/embodiment/openpi_rlinf/transforms_pipeline.py:74]: Thanks. One clarification: in the current implementation, explicit_norm_stats_path is normalized and assigned to norm_stats_dir before resolution, so it overrides a separately supplied norm_stats_dir. The actual priority is:
data_config.norm_stats > explicit_norm_stats_path > norm_stats_dir > download.
I updated the function docstring to reflect the existing behavior without changing the resolution logic.

File: examples/embodiment/config/maniskill_rlt_stage2_ac_mlp.yaml (hunk 4 above)
Comment (TihayaKousaka) [reply examples/embodiment/config/maniskill_rlt_stage2_ac_mlp.yaml]: fixed
